DOP-4928 - update docs search page for action bar

seungpark opened the pull request:
### Stories/Links:

DOP-4928

[Figma Designs](https://www.figma.com/design/7cR6eWtFvKTnJIPL8ZsTaB/Content-Template-Projects?node-id=6642-20413&node-type=FRAME&t=6prYsC2YXvJcBdqB-0)

- This PR updates search results page to conform to new Action Bar Universal Search.
  - Removes existing search input on search page
  - Updates styling to match with action bar spacing

### Current Behavior:

[current search results](https://www.mongodb.com/docs/search/?q=aggregation&page=1)

### Staging Links:

[staging link with updated search results](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/seung.park/DOP-4928/search/index.html?q=aggregation&page=1)
[updated spacing on 404 page](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/docs-404/seung.park/DOP-4928/index.html)

### Notes:

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

seungpark commented on src/components/ActionBar/SearchMenu.js:
```diff
@@ -1,31 +1,52 @@
+  const { project } = useSnootyMetadata();
+  const docsets = useAllDocsets();
+  const { snootyEnv } = useSiteMetadata();
+  const fullSearchUrl = useMemo(() => {
```
i'd like to test this in stage/qa since we don't have a valid `development` env here. open to suggestions for fallbacks

seungpark replied:
tested this and the localization on stage. (while stage is being tested), you can see it on docs-qa

https://mongodbcom-cdn.website.staging.corp.mongodb.com/ko-kr/docs-qa/atlas/

searching from here leads you to https://mongodbcom-cdn.website.staging.corp.mongodb.com/ko-kr/docs-qa/search/?q=test

mmeigs reviewed (commented):
About to look over this! But I never stop staring at the annoying HALF pixel that the border on the sidenav and the border on the actionBar don't line up. It's a silliness of our old css. I finally kinda figured it out.

In Sidenav.js can you change the first two lines of CSS in the Border (styled.hr) to this: 
```
  border: 1px solid transparent;
  border-bottom: 1px solid var(--sidenav-border-bottom-color);
``` 

Having the border as unset actually put it off by half a pixel lol

mmeigs commented:
Oh, quick note: the spacing of the ActionBar is incorrect (I think) now on the legacy page
<img width="1161" alt="Screenshot 2024-09-03 at 2 46 37 PM" src="https://github.com/user-attachments/assets/66807c3c-539a-4e6f-a71a-694fa3fd8357">

seungpark commented:
> About to look over this! But I never stop staring at the annoying HALF pixel that the border on the sidenav and the border on the actionBar don't line up. It's a silliness of our old css. I finally kinda figured it out.
> 
> In Sidenav.js can you change the first two lines of CSS in the Border (styled.hr) to this:
> 
> ```
>   border: 1px solid transparent;
>   border-bottom: 1px solid var(--sidenav-border-bottom-color);
> ```
> 
> Having the border as unset actually put it off by half a pixel lol

discussed offline. this was due to zoom levels within the browser. dropped in a ticket to address this issue https://jira.mongodb.org/browse/DOP-4977

mayaraman19 commented on src/components/ActionBar/SearchMenu.js:
```diff
@@ -1,31 +1,58 @@
 // to expose lazy loaded child (chatbot) behaviors to parent (SearchInput)
 // https://react.dev/reference/react/useImperativeHandle
-const SearchMenu = forwardRef(function SearchMenu({ searchValue, searchBoxRef, isOpen, selectedOption }, ref) {
+const SearchMenu = forwardRef(function SearchMenu({ searchValue, searchBoxRef, isOpen, selectedOption, slug }, ref) {
```
if you want to avoid prop drilling `slug` you can use [PageContext](https://github.com/mongodb/snooty/blob/main/src/context/page-context.js) which contains the slug value

seungpark replied:
I've actually tried this! but this PageContext is not in scope of the actionbar.

ActionBar is defined outside of the DocumentBody template https://github.com/mongodb/snooty/blob/DOP-4928/src/layouts/index.js#L134

and the PageContext is defined within the DocumentBody https://github.com/mongodb/snooty/blob/main/src/components/DocumentBody.js#L153

mayaraman19 reviewed (commented):
search page looks good! minor comment below.

also, not in the scope of this ticket but i want to note it somewhere - is the dark mode toggle logo too small? in the figma it is 28 x 28 pixels, but ours is 16 x 16.

<img width="153" alt="image" src="https://github.com/user-attachments/assets/0b88a5a1-a394-4a9e-99c5-34cd9f5ce649">

also looks like the "system" icon is a different shade of gray than the "dark mode" icon.

seungpark commented:
> search page looks good! minor comment below.
> 
> also, not in the scope of this ticket but i want to note it somewhere - is the dark mode toggle logo too small? in the figma it is 28 x 28 pixels, but ours is 16 x 16.
> 
> <img alt="image" width="153" src="https://private-user-images.githubusercontent.com/41043008/364431665-0b88a5a1-a394-4a9e-99c5-34cd9f5ce649.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0NjI1OTYsIm5iZiI6MTcyNTQ2MjI5NiwicGF0aCI6Ii80MTA0MzAwOC8zNjQ0MzE2NjUtMGI4OGE1YTEtYTM5NC00YTllLTk5YzUtMzRjZDlmNWNlNjQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA0VDE1MDQ1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkM2Y5NjQ3MWExZjQ3MjcxM2MyM2U4ODM5YzgwMDJjYWJlZDJmMjE4NThhMGViMjljZDIwYjE0MzQ0NTZkNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VJaLeUEhknrkx_s8QWnpO_fMWKAPJCDN2MRID15lkPE">
> also looks like the "system" icon is a different shade of gray than the "dark mode" icon.

Actually the system icon is going away! and the icons sizing needs to be updated https://jira.mongodb.org/browse/DOP-4968

seungpark commented:
> Oh, quick note: the spacing of the ActionBar is incorrect (I think) now on the legacy page
> 
> <img alt="Screenshot 2024-09-03 at 2 46 37 PM" width="1161" src="https://private-user-images.githubusercontent.com/22421112/364102671-66807c3c-539a-4e6f-a71a-694fa3fd8357.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0ODU4OTEsIm5iZiI6MTcyNTQ4NTU5MSwicGF0aCI6Ii8yMjQyMTExMi8zNjQxMDI2NzEtNjY4MDdjM2MtNTM5YS00ZTZmLWE3MWEtNjk0ZmEzZmQ4MzU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA0VDIxMzMxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmMmM4YjBkNmE5ZjE3ZTYxNjE0YjIzMWJkZTNjOTQ1ZTYxMGRlMzM1ZDRhOWIxMDJkNDBkMzYzYzZhNzIzNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WaS123MkaMVCwZm23Cp3e1FX2VPzu2lgnfG2xjRkHFo">

Thanks for flagging this @mmeigs. I updated the legacy page to use a grid layout, similar to the 404 page. I've updated the staging link and added 404 page to double check

> Actually the system icon is going away! and the icons sizing needs to be updated https://jira.mongodb.org/browse/DOP-4968

@mayaraman19 [working branch](https://github.com/mongodb/snooty/compare/DOP-4616...DOP-4968-update-darkmode-icons) until icons are finalized from design!

mmeigs reviewed (approved):
LGTM!
